### DISCUSSION TITLE
Performance issues with multiple components

### DISCUSSION THREAD

**Discussion by HashNotAdam** [unabridged/motion#83] [Q&A]
I've found that having a few motion components on a page quickly overwhelms the system and the webserver becomes unresponsive. To get around the issue I've been using JavaScript to send events up to a singular motion listener (for example, maybe update a data value and click a button or add a hidden value to a form). This really goes against the point of using motion.

Has anyone found a solution to this issue?

**Comment by latortuga** [comment]
Hi @HashNotAdam, good morning.

I'm interested to know more about this scenario and the types of motions you're using. Can you give some more specifics about the kind of traffic you're under, your app deployment setup, and the number of motions you're processing per minute?

I'm confident we can figure out a solution to your issue, and in motion if it turns out to be a motion issue!

We are currently using Motion in production and haven't had this experience so I'm keen to uncover scaling issues that we can be proactive about.

**Reply by HashNotAdam** [reply]
I've had the problem with only 2 users (requires more than 1) but I can't replicate it in a minimal app. I'm going to need to find time to remove my JavaScript hack and try to eliminate variables.
